x-pack/plugins/apm/public/components/alerting/TransactionErrorRateAlertTrigger/index.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiFieldNumber, EuiSelect, EuiExpression } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ForLastExpression } from '../../../../../triggers_actions_ui/public';
+import { ALERT_TYPES_CONFIG, AlertType } from '../../../../common/alert_types';
+import { useEnvironments } from '../../../hooks/useEnvironments';
+import { useServiceTransactionTypes } from '../../../hooks/useServiceTransactionTypes';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { ServiceAlertTrigger } from '../ServiceAlertTrigger';
+import { PopoverExpression } from '../ServiceAlertTrigger/PopoverExpression';
+import {
+  ENVIRONMENT_ALL,
+  getEnvironmentLabel,
+} from '../../../../common/environment_filter_values';
+
+interface AlertParams {
+  windowSize: number;
+  windowUnit: string;
+  threshold: number;
+  serviceName: string;
+  transactionType: string;
+  environment: string;
+}
+
+interface Props {
+  alertParams: AlertParams;
+  setAlertParams: (key: string, value: any) => void;
+  setAlertProperty: (key: string, value: any) => void;
+}
+
+export function TransactionErrorRateAlertTrigger(props: Props) {
+  const { setAlertParams, alertParams, setAlertProperty } = props;
+  const { urlParams } = useUrlParams();
+  const transactionTypes = useServiceTransactionTypes(urlParams);
+  const { serviceName, start, end, transactionType } = urlParams;
```

Comment:
In other triggers we use

`const { serviceName } = useParams<{ serviceName?: string }>();`

Should it be consistent?

Reply:
Fixed